File: src/main/kotlin/org/kite9/diagram/visualization/planarization/PlanarizationException.java
package org.kite9.diagram.visualization.planarization;

import org.kite9.diagram.logging.LogicException;

public class PlanarizationException extends LogicException {

	private static final long serialVersionUID = 3493217216660687045L;

	Planarization pln;
	
	public PlanarizationException(String string, Planarization pln, Exception e) {
		super(string, e);
		this.pln = pln;
	}
	
	public Planarization getPlanarization() {
		return pln;
	}

}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/PlanarizationException.kt
package org.kite9.diagram.visualization.planarization

import org.kite9.diagram.model.Diagram
import org.kite9.diagram.visualization.planarization.Planarization
import org.kite9.diagram.logging.LogicException
import java.lang.Exception

class PlanarizationException(string: String?, var planarization: Planarization, e: Exception?) :
    LogicException(string, e) {

    companion object {
        private const val serialVersionUID = 3493217216660687045L
    }
}